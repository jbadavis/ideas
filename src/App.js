import React from "react";
import Idea from "./Idea";

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      ...this.props.initialState
    };
  }

  render() {
    return (
      <div className="App">
        <h1>Ideas</h1>
        {this.state.ideas.map((idea, i) => (
          <Idea
            key={i}
            {...idea}
          />
        ))}
        <button>Add Idea</button>
      </div>
    );
  }
}

export default App;
