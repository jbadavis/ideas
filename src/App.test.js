import React from 'react';
import App from './App';
import {shallow} from 'enzyme';

describe('App', () => {
  const initialStateMock = {
    ideas: [
      {
        title: 'title one',
        description: 'description one',
      },
      {
        title: 'title one',
        description: 'description one',
      },
    ],
  };

  it('should match snapshot', () => {
    expect(shallow(<App initialState={initialStateMock} />)).toMatchSnapshot();
  });
});
