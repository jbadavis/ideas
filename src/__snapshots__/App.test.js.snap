// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Ideas
  </h1>
  <Idea
    description="description one"
    key="0"
    title="title one"
  />
  <Idea
    description="description one"
    key="1"
    title="title one"
  />
  <button>
    Add Idea
  </button>
</div>
`;
